src/index.ts

```diff
@@ -98,22 +121,72 @@ const memoryOnlyPrecomputedBanditsStore = precomputedBanditStoreFactory();
  * @public
  */
 export class EppoJSClient extends EppoClient {
-  // Ensure that the client is instantiated during class loading.
-  // Use an empty memory-only configuration store until the `init` method is called,
-  // to avoid serving stale data to the user.
+  /**
+   * Resolved when the client is initialized
+   * @private
+   */
+  private readonly readyPromise: Promise<void>;
+
+  /**
+   * Used when the client is initialized from outside the constructor, namely, from the `init` or
+   * `EppoJSClient.initializeClient` methods.
+   * @private
+   */
+  private readyPromiseResolver: (() => void) | null = null;
+
   public static instance = new EppoJSClient({
     flagConfigurationStore,
     isObfuscated: true,
   });
-  public static initialized = false;
+  initialized = false;
+
+  constructor(optionsOrConfig: EppoClientParameters | IClientOptions) {
+    const v2Constructor = 'sdkKey' in optionsOrConfig;
+
+    super(
+      v2Constructor
+        ? clientOptionsToEppoClientParameters(
+            optionsOrConfig, // The IClientOptions wrapper
+            optionsOrConfig.flagConfigurationStore ?? // create a new, memory only FCS if none was provided
+              configurationStorageFactory({
+                forceMemoryOnly: true,
+              }),
+            optionsOrConfig.eventIngestionConfig // Create an Event Dispatcher if Event Ingestion config was provided.
+              ? newEventDispatcher(optionsOrConfig.sdkKey, optionsOrConfig.eventIngestionConfig)
+              : undefined,
+          ) // "New" construction technique; isolated instance.
+        : (optionsOrConfig as EppoClientParameters), // Legacy instantiation of singleton.
+    );
+
+    if (!v2Constructor) {
+      // Original constructor path; passthrough to super is above.
+      // `waitForReady` is a new API we need to graft onto the old way of constructing first and initializing later.
+
+      // Create a promise that will be resolved when EppoJSClient.initializeClient() is called
+      this.readyPromise = new Promise((resolve) => {
+        this.readyPromiseResolver = resolve;
+      });
+    } else {
+      this.readyPromise = explicitInit(
+        convertClientOptionsToClientConfig(optionsOrConfig),
+        this,
+      ).then(() => {
+        return;
+      });
+    }
+  }
+
+  public waitForReady(): Promise<void> {
```

Comment:
new method! 🎉

Reply:
what does "ready" mean in this context? adding comments is likely to help, also we should always do it for public APIs anyways

Reply:
`waitForConfiguration` to be in line with multiplatform's analog `PollerThread::waitForConfiguration`